src/analysis/retail/mage/arcane/core/ArcaneBarrage.tsx

```diff
@@ -112,13 +120,31 @@ export default class ArcaneBarrage extends Analyzer {
 
     this.arcaneChargeTracker.clearCharges(event);
   }
+
+suggestions() {
+  this.allCasts.sort((a, b) => a.timestamp - b.timestamp);
+  for (let i = 0; i < this.barrageCasts.length; i++) {
+    const barrage = this.barrageCasts[i];
+    const barrageIndex = this.allCasts.findIndex(
+      c => c.timestamp === barrage.cast.timestamp,
+    );
```

Comment:
this method is quadratic in the number of arcane barrages cast. you likely won't see a slowdown on raid fights, but in dungeons this can cause significant performance problems.

the appropriate way to implement this is do a "pending" system.

1. when creating a `barrageCast` object, set a flag in this module like `this.checkNextCast = true`
2. in your `onAnyCast` handler, check it and update the most recent barrage, like so:

```tsx
onAnyCast(event: CastEvent) {
  if (this.checkNextCast) {
    const barrage = this.barrageCasts[this.barrageCasts.length - 1];
    barrage.nextCastIsTouch = event.ability.guid === TALENTS.TOUCH_OF_THE_MAGI_TALENT.id;
  }
  this.checkNextCast = false;
} 
```

then you can entirely remove this method.